apps/web/pages/category/[slug].vue

```diff
@@ -32,40 +46,36 @@ const { getFacetsFromURL } = useCategoryFilter();
 const { fetchProducts, data: productsCatalog, productsPerPage } = useProducts();
 const { getCategoryTree, data: categoryTree } = useCategoryTree();
 const { t } = useI18n();
-
-const findCategoryBySlugs = (categories: CategoryTreeItem[], slugs: string[]): CategoryTreeItem | undefined => {
-  const category = categories.find((item) => categoryTreeGetters.getSlug(item) === slugs[0]);
-  if (category && slugs.length > 1 && category.children) {
-    return findCategoryBySlugs(category.children, slugs.slice(1));
-  }
-  return category;
-};
+const breadcrumbs = ref([] as Breadcrumb[]);
+const urlParams = ref(getFacetsFromURL());
+const category = ref(
+  categoryTreeGetters.findCategoryBySlugs(categoryTree.value, urlParams.value.categorySlugs || ['']),
+);
+const parentCategory = ref(
+  categoryTreeGetters.findCategoriesPathByCategoryId(categoryTree.value, category.value?.id ?? 0),
+);
 
 const generateSearchParams = () => {
-  const urlParams = getFacetsFromURL();
-  const category = findCategoryBySlugs(categoryTree.value, urlParams.categorySlugs || ['']);
-  urlParams.categoryId = category?.id?.toString();
-  return urlParams;
+  urlParams.value = getFacetsFromURL();
+  category.value = categoryTreeGetters.findCategoryBySlugs(categoryTree.value, urlParams.value.categorySlugs || ['']);
+  urlParams.value.categoryId = category.value?.id?.toString();
+  parentCategory.value = categoryTreeGetters.findCategoriesPathByCategoryId(
+    categoryTree.value,
+    category.value?.id ?? 0,
+  );
+
+  breadcrumbs.value = category.value
+    ? categoryTreeGetters.generateBreadcrumbFromCategory(categoryTree.value, category.value)
+    : [{ name: t('allProducts'), link: '#' }];
+
+  breadcrumbs.value.unshift({ name: t('home'), link: '/' });
```

Comment:
I get why you moved this into the function, but it has nothing to do with the search params right?
Only that you update them once something changes. So maybe also move it into a function. How do you feel about this?